velox/dwio/common/SelectiveColumnReader.h

```diff
@@ -26,6 +26,13 @@
 
 namespace facebook::velox::dwio::common {
 
+#define VELOX_DECODE_WITH_VISITOR(decoder, nulls, visitor) \
```

Comment:
Make this a private method of column reader, and since `nulls` is always from `nullsInReadRange_`, no need to pass it by parameter

Reply:
Do you mean that we should replace the VELOX_DECODE_WITH_VISITOR macro with a protected method of SelectiveColumnReader (because nullsInReadRange_ defined here)?

To do this, we should make DirectDecoder::readWithVisitor as a virtual function of IntDecoder, or move RLEv2.h and RLEv1.h to dwio/common (just like DirectDecoder.h), do you think that's OK? and which one is more reasonable?